### PR TITLE
feat(j-s): JWT verification for digital-mailbox-api

### PR DESCRIPTION
**Pull request by oddsson**
# JWT verification for digital-mailbox-api

[Asana](https://app.asana.com/0/1199153462262248/1207323461913719/f)

## What

Verify JWT sent with requests to digital-mailbox-api. 

## Why

So that only people authenticated on island.is can call this api.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
  - Enhanced authentication configuration for better security and flexibility.
  - Added token verification for user authentication.

- **Improvements**
  - Updated authentication settings including scope, client ID, issuer URL, and redirect URIs.
  - Improved connection testing with token verification.

- **Bug Fixes**
  - Fixed issues related to authentication bypass and backend URL configuration.

### CODE REVIEW

File: apps/judicial-system/digital-mailbox-api/src/app/app.config.ts [hunk 1]
```diff
@@ -1,15 +1,28 @@
 import { defineConfig } from '@island.is/nest/config'
 
-export default defineConfig({
-  name: 'AppModule',
+export const authModuleConfig = defineConfig({
+  name: 'AuthModule',
```
Comment (unakb) [inline apps/judicial-system/digital-mailbox-api/src/app/app.config.ts:4]: I think we should name this something other than AuthModule (and AppModule to be fair wasn't the best name either). Maybe just DigitalMailboxModule or something like that

File: apps/judicial-system/digital-mailbox-api/src/app/app.service.ts [hunk 2]
```diff
@@ -21,8 +24,54 @@ export class AppService {
...
+    const verifiedUserToken = await this.verifyIdsToken(token)
+
+    if (!verifiedUserToken.sub) {
+      throw new UnauthorizedException('Invalid token')
+    }
+
     return this.test()
   }
+
+  async verifyIdsToken(token: string) {
```
Comment (unakb) [inline apps/judicial-system/digital-mailbox-api/src/app/app.service.ts:38]: I think we should move this to a guard so it's easier to use for all the endpoints

Comment (unakb) [reply apps/judicial-system/digital-mailbox-api/src/app/app.service.ts:38]: We could start with something along the lines of this 
https://github.com/island-is/island.is/commit/bfb45f65bd6f01f957cc222108066b8f2f6e9c79

I don't know whether we want to have this in our lib or in this project, but could be a start to just have it here

Comment (unakb) [reply apps/judicial-system/digital-mailbox-api/src/app/app.service.ts]: I also realized we can use the passport library to handle the verification, a draft of the idea :
https://github.com/island-is/island.is/pull/14882

Should we start with something like this? It also makes it easier to extract the national id to verify that the user can access cases only connected to their national id

**Comment by oddsson** [comment]
Closed in favour of https://github.com/island-is/island.is/pull/14882

**Pull request closed without merging**